### PR TITLE
Fix the spans of catch blocks to include the `do`

### PR DESCRIPTION
**Pull request by nrc**

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -2243,9 +2243,9 @@ impl<'a> Parser<'a> {
                         attrs);
                 }
                 if self.is_catch_expr() {
+                    let lo = self.span;
```
Comment (topecongiro) [inline src/libsyntax/parse/parser.rs]: Shouldn't this come after `self.eat_keyword(keywords::Do)` instead of before?

Comment (nrc) [inline src/libsyntax/parse/parser.rs]: I changed from using `prev_span` to `span`, so we can move it before the eat(Do) expression

Comment (topecongiro) [inline src/libsyntax/parse/parser.rs]: Sorry, I missed the change. Thank you for your reply, and sorry for the noise.

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
r=me with a test; the most straightforward thing would be to add a `src/test/ui` test that has incorrect highlighting with the current span.

**Pull request merged**